pkg/metrics/metrics.go

```diff
@@ -0,0 +1,147 @@
+package metrics
+
+import (
+	"net/http"
+	"sync"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/klog/v2"
+)
+
+var (
+	// global instance of MetricRecorder
+	r *MetricRecorder = NewRecorder()
+)
+
+type MetricRecorder struct {
+	registry metrics.KubeRegistry
+	register sync.Once
+
+	apiDuration   *metrics.HistogramVec
+	apiError      *metrics.CounterVec
+	apiThrottle   *metrics.CounterVec
+	internalValue *metrics.HistogramVec
+	internalError *metrics.CounterVec
+}
+
+// Recorder returns the global instance of MetricRecorder
+func Recorder() *MetricRecorder {
+	return r
+}
+
+// NewRecorder initializes a new MetricRecorder with default metric definitions.
+func NewRecorder() *MetricRecorder {
+	return &MetricRecorder{
+		registry: metrics.NewKubeRegistry(),
+		apiDuration: metrics.NewHistogramVec(
+			&metrics.HistogramOpts{
+				Name:           "cloudprovider_aws_api_request_duration_seconds",
+				Help:           "Latency of AWS API calls",
+				StabilityLevel: metrics.ALPHA,
+			},
+			[]string{"request"},
+		),
+		apiError: metrics.NewCounterVec(
+			&metrics.CounterOpts{
+				Name:           "cloudprovider_aws_api_request_errors",
+				Help:           "AWS API errors",
+				StabilityLevel: metrics.ALPHA,
+			},
+			[]string{"request"},
+		),
+		apiThrottle: metrics.NewCounterVec(
+			&metrics.CounterOpts{
+				Name:           "cloudprovider_aws_api_throttled_requests_total",
+				Help:           "AWS API throttled requests",
+				StabilityLevel: metrics.ALPHA,
+			},
+			[]string{"operation_name"},
+		),
+		internalValue: metrics.NewHistogramVec(
```

Comment:
Nit: Rename internalValue to something like csiOperationDuration?

Reply:
Better question, what is internalValue? And if it's not measuring csiOperationDuration, why aren't we measuring it?

Reply:
`internalValue` here was meant to be a generic metric that could be used for capturing both duration and values in a histogram -- latest rev reworked things quite a bit and incorporates this feedback.